invoice/src/amount.rs

```diff
@@ -259,6 +268,41 @@ impl CoinAmount {
             precision,
         }
     }
+
+    pub fn to_value(self) -> u64 {
```

Comment:
I think we need to avoid using naive `u64` since it is not clear what semantics are associated with it (fractional, integer part or the whole atomic value).

```suggestion
    pub fn to_amount(&self) -> Amount {
```

Reply:
Actually, the correct signature should be
```suggestion
    pub fn precise_amount(self, precision: Precision) -> Amount {
```

I mean if we move to `CoinAmount::min_precision`, we can't construct Amount without exact precision coming from the external contract information.